File: Algorithms/Dynamic_Programming/Golomb Sequence/Golomb.java
import java.util.*; 
  
class Golomb  
{ 
      
    public static void printGolomb(int n) 
    { 
        int dp[] = new int[n + 1];  
        dp[1] = 1; 
        System.out.print(dp[1] + " "); 
        for (int i = 2; i <= n; i++)  
        { 
            dp[i] = 1 + dp[i - dp[dp[i - 1]]]; 
              
        System.out.print(dp[i] + " "); 
        } 
    } 
    public static void main (String[] args) 
    { 
        Scanner s = new Scanner(System.input());
        int n = s.nextInt();
        printGolomb(n); 
    } 
}

File: Algorithms/Dynamic_Programming/Golomb_Sequence/golomb.py
# Golomb sequence 
def findGolomb(n): 

	if (n == 1): 
		return 1

	return 1 + findGolomb(n - findGolomb(findGolomb(n - 1))) 
 
def printGolomb(n): 

	for i in range(1, n + 1): 
		print(findGolomb(i), end=" ") 

n = 19
printGolomb(n) 
